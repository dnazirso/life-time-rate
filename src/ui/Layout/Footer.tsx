import {
  CalendarViewWeek,
  CalendarToday,
  PermContactCalendar,
} from "@mui/icons-material";
import AccountCircleIcon from '@mui/icons-material/AccountCircle';
import { BottomNavigation, BottomNavigationAction, Box } from "@mui/material";
import { useNavigate } from "react-router-dom";
import { useDispatch } from "react-redux";
import { Page, setPage } from "../../core/Store/appSlice";
import { useAppSelector } from "../../core/Store";

export default function Footer({ signed }: { signed: boolean }) {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const { page } = useAppSelector((state) => state.app);

  const handleChange = (
    _: React.SyntheticEvent<Element, Event>,
    name: Page
  ) => {
    dispatch(setPage(name));
    navigate(name);
  };

  return (
    <Box
      component={BottomNavigation}
      showLabels
      value={page}
      onChange={handleChange}
      display={signed ? "flex" : "none"}
    >
      <BottomNavigationAction
        label="Years"
        value="/years"
        icon={<PermContactCalendar />}
      />
      <BottomNavigationAction
        label="Weeks"
        value="/weeks"
        icon={<CalendarViewWeek />}
      />
      <BottomNavigationAction
        label="Today"
        value="/today"
        icon={<CalendarToday />}
      />
      <BottomNavigationAction
        label="Profile"
        value="/profile"
        icon={<AccountCircleIcon />}
      />
    </Box>
  );
}
