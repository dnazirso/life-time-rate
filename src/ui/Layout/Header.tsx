import { Logout } from "@mui/icons-material";
import {
  AppBar,
  Container,
  IconButton,
  Toolbar,
  Typography,
} from "@mui/material";
import { useContext } from "react";
import { useNavigate } from "react-router-dom";
import SignInContext from "../../core/SignInContext";
import { useAppDispatch, useAppSelector } from "../../core/Store";
import { setPage } from "../../core/Store/appSlice";

export default function Header({ signed }: { signed: boolean }) {
  const navigate = useNavigate();
  const dispatch = useAppDispatch();
  const context = useContext(SignInContext);
  const { title } = useAppSelector((state) => state.app);

  const handleLogout = () => {
    context.logout();
    dispatch(setPage("/"));
    navigate("/");
  };

  return (
    <AppBar position="sticky">
      <Container>
        <Toolbar disableGutters>
          <Typography component="div" variant="h5" sx={{ flexGrow: 1 }} px={2}>
            {title || "Life Calendar"}
          </Typography>
          <IconButton
            size="large"
            sx={{ display: signed ? "" : "none" }}
            onClick={handleLogout}
          >
            <Logout />
          </IconButton>
        </Toolbar>
      </Container>
    </AppBar>
  );
}
