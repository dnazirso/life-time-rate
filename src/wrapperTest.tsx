import { ReactNode } from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { store } from "./core/Store";
import ThemeContextProvider from "./ui/App/ThemeContextProvider";

export function wrapper({ children }: { children: ReactNode }) {
  return (
    <Provider store={store}>
      <ThemeContextProvider>
        <BrowserRouter>{children}</BrowserRouter>
      </ThemeContextProvider>
    </Provider>
  );
}
