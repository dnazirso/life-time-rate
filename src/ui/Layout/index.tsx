import { ReactNode, useContext } from "react";
import { CssBaseline, Grid } from "@mui/material";
import Header from "./Header";
import Footer from "./Footer";
import SignInContext from "../../core/SignInContext";

export default function Layout({
  children,
  title,
}: {
  children: ReactNode;
  title?: string;
}) {
  const context = useContext(SignInContext);

  const signed = context.user.birthdate !== new Date("10/10/1910").getTime();

  return (
    <Grid container direction="column" minHeight="100vh">
      <CssBaseline />
      <Grid item>
        <Header signed={signed} />
      </Grid>
      <Grid item container justifyContent="center" alignItems="center" xs>
        <Grid item>{children}</Grid>
      </Grid>
      <Grid item>
        <Footer signed={signed} />
      </Grid>
    </Grid>
  );
}
