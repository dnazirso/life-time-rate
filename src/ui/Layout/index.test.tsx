import { render } from "@testing-library/react";
import { wrapper } from "../../wrapperTest";

import Layout from "./";

describe(`${Layout.name}`, () => {
  it("renders correctly", () => {
    // Arrange
    const { asFragment } = render(<Layout>TEST</Layout>, { wrapper });

    // Act
    const result = asFragment();

    // Assert
    expect(result).toMatchSnapshot();
  });
});
