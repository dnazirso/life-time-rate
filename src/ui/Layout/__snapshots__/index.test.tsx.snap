// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1e73fxf-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionSticky css-7e0s21-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular css-1cndkqf-MuiToolbar-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h5 css-1l8p2q8-MuiTypography-root"
            >
              Life calendar
            </div>
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-dpjt1-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="LogoutIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m17 7-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </header>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-true css-1r61agb-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        TEST
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiBottomNavigation-root MuiBox-root css-bvmudu-MuiBottomNavigation-root"
      >
        <button
          class="MuiButtonBase-root MuiBottomNavigationAction-root css-1ee5err-MuiButtonBase-root-MuiBottomNavigationAction-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="PermContactCalendarIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3h-1V1h-2v2H8V1H6v2H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm6 12H6v-1c0-2 4-3.1 6-3.1s6 1.1 6 3.1v1z"
            />
          </svg>
          <span
            class="MuiBottomNavigationAction-label css-imwso6-MuiBottomNavigationAction-label"
          >
            Years
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiBottomNavigationAction-root css-1ee5err-MuiButtonBase-root-MuiBottomNavigationAction-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CalendarViewWeekIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm-7 2h2.5v12H13V6zm-2 12H8.5V6H11v12zM4 6h2.5v12H4V6zm16 12h-2.5V6H20v12z"
            />
          </svg>
          <span
            class="MuiBottomNavigationAction-label css-imwso6-MuiBottomNavigationAction-label"
          >
            Weeks
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiBottomNavigationAction-root css-1ee5err-MuiButtonBase-root-MuiBottomNavigationAction-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CalendarTodayIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 18H4V8h16v13z"
            />
          </svg>
          <span
            class="MuiBottomNavigationAction-label css-imwso6-MuiBottomNavigationAction-label"
          >
            Today
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
